src/main/java/seedu/address/model/Database.java

```diff
@@ -228,6 +236,41 @@ public boolean equals(Object other) {
                 && patients.equals(otherDatabase.patients);
     }
 
+    /**
+     * Checks if the appointment is valid to be inserted into the appointment list
+     * @throws CommandException if {@code predicate} is null.
+     */
+    void validateAddAppointment(Appointment reference, Appointment toAdd) throws CommandException {
+        for (Appointment a : appointments) {
+            if (toAdd.getPatientNric().equals(a.getPatientNric())
+                    && (toAdd.overlaps(a))
+                    && !a.equals(reference)) {
+                throw new CommandException(MESSAGE_OVERLAPPING_PATIENT_APPOINTMENTS);
+            }
+            if (toAdd.getDoctorNric().equals(a.getDoctorNric())
+                    && (toAdd.overlaps(a))
+                    && !a.equals(reference)) {
+                throw new CommandException(MESSAGE_OVERLAPPING_DOCTOR_APPOINTMENTS);
+            }
+        }
+    }
+
+    /**
+     * Checks if the appointment is valid to be inserted into the appointment list
+     * @throws CommandException if {@code predicate} is null.
+     */
+    void validateAddAppointment(Appointment toAdd) throws CommandException {
+        for (Appointment a : appointments) {
```

Comment:
dont duplicate the code. this method should call `validateAddAppointment(null, toAdd)` instead